src/controllers/places/modify_content.js

```diff
@@ -44,43 +44,60 @@ module.exports = (req, res) => {
     if (err) {
       return res.status(500).send(res.locals.localText.serverError);
     }
-    let url, urlEndpoint, correctResponseStatusCode, auth;
+
+    let headers;
+    const tools = {};
     switch (req.body._method) {
       case 'post':
-        url = placesURL;
-        urlEndpoint = 'places';
-        correctResponseStatusCode = 201;
+        tools.url = placesURL;
+        tools.urlEndpoint = 'places';
+        tools.correctResponseStatusCode = 201;
         break;
       case 'put':
-        url = `${placesURL}/${req.params.id}`;
-        urlEndpoint = `place/${req.params.id}`;
-        correctResponseStatusCode = 200;
-        auth = {
-          bearer: decodedToken,
+        tools.url = `${placesURL}/${req.params.id}`;
+        tools.urlEndpoint = `PLACE/${req.params.id}`;
+        tools.correctResponseStatusCode = 200;
+        headers = {
+          Authorization: 'Bearer ' + decodedToken,
         };
         break;
       default:
         return res.status(500).send(res.locals.localText.serverError);
     }
+
+    // adds the redirectUrls to the tools
+    tools.redirectUrl = JSON.stringify({
+      redirectUrl: `/${req.params.lang}/${tools.urlEndpoint}`,
+    });
+
     const reqOptions = {
-      url,
+      url: tools.url,
       method: req.body._method,
-      body: apiBody,
-      json: true,
-      auth,
+      data: apiBody,
+      responseType: 'json',
+      headers,
     };
-    Request(reqOptions, (error, apiResponse, apiResponseBody) => {
-      if (error) {
-        return res.status(500).send(res.locals.localText.serverError);
-      } else if (apiResponse.statusCode !== correctResponseStatusCode) {
-        return res
-          .status(apiResponseBody.statusCode)
-          .send(apiResponseBody.message);
-      } else {
-        res.end(
-          JSON.stringify({ redirectUrl: `/${req.params.lang}/${urlEndpoint}` }),
-        );
-      }
-    });
+
+    OTP.events
+      .modify(reqOptions, tools)
```

Comment:
okay .... ..

I am looking here: https://github.com/foundersandcoders/OTP-Data-Entry/blob/master/src/otp_sdk/events.js

I would appear that I missed a couple of problems.

  1. What is tools?? : https://github.com/foundersandcoders/OTP-Data-Entry/blob/master/src/otp_sdk/events.js see here it is not being used.

The name `tools` is also totally wack..

`OTP.events.modify` has 2 problems, first, it should not take a url, that should be defined by the otpsdk. OTP.events.modify should take a blob of json (the update) and the users auth token.

I think it would be better to call `OTP.events.modify` `OTP.events.update`

Last (but not least) I think this file is updating a place, not an event .